x/observer/keeper/vote_outbound_test.go

```diff
@@ -0,0 +1,193 @@
+			sample.AccAddress(),
+		)
+		require.Error(t, err)
+		require.ErrorIs(t, err, types.ErrNotObserver)
+	})
+
+	t.Run("can add vote and create ballot", func(t *testing.T) {
+		k, ctx, _ := keepertest.ObserverKeeperWithMocks(t, keepertest.ObserverMocksAll)
+
+		observer := sample.AccAddress()
+		stakingMock := keepertest.GetObserverStakingMock(t, k)
+		slashingMock := keepertest.GetObserverSlashingMock(t, k)
+
+		k.SetChainParamsList(ctx, types.ChainParamsList{
+			ChainParams: []*types.ChainParams{
+				{
+					ChainId:     getValidEthChainIDWithIndex(t, 0),
+					IsSupported: true,
+				},
+			},
+		})
+		k.SetObserverSet(ctx, types.ObserverSet{
+			ObserverList: []string{observer},
+		})
+		stakingMock.MockGetValidator(sample.Validator(t, sample.Rand()))
+		slashingMock.MockIsTombstoned(false)
+
+		isFinalized, isNew, ballot, _, err := k.VoteOnOutboundBallot(
+			ctx,
+			"index",
+			getValidEthChainIDWithIndex(t, 0),
+			common.ReceiveStatus_Success,
+			observer,
+		)
+		require.NoError(t, err)
+
+		// ballot should be finalized since there is only one observer
+		require.True(t, isFinalized)
+		require.True(t, isNew)
+		expectedBallot, found := k.GetBallot(ctx, "index")
+		require.True(t, found)
+		require.Equal(t, expectedBallot, ballot)
+	})
+
+	t.Run("can add vote to an existing ballot", func(t *testing.T) {
+		k, ctx, _ := keepertest.ObserverKeeperWithMocks(t, keepertest.ObserverMocksAll)
+
+		observer := sample.AccAddress()
+		stakingMock := keepertest.GetObserverStakingMock(t, k)
+		slashingMock := keepertest.GetObserverSlashingMock(t, k)
+
+		k.SetChainParamsList(ctx, types.ChainParamsList{
+			ChainParams: []*types.ChainParams{
+				{
+					ChainId:     getValidEthChainIDWithIndex(t, 0),
+					IsSupported: true,
+				},
+			},
+		})
+		k.SetObserverSet(ctx, types.ObserverSet{
+			ObserverList: []string{observer},
+		})
+		stakingMock.MockGetValidator(sample.Validator(t, sample.Rand()))
+		slashingMock.MockIsTombstoned(false)
+
+		// set a ballot
+		threshold, err := sdk.NewDecFromStr("0.7")
+		require.NoError(t, err)
+		ballot := types.Ballot{
+			Index:            "index",
+			BallotIdentifier: "index",
+			VoterList: []string{
+				sample.AccAddress(),
+				sample.AccAddress(),
+				observer,
+				sample.AccAddress(),
+				sample.AccAddress(),
+			},
+			Votes:           types.CreateVotes(5),
+			ObservationType: types.ObservationType_OutBoundTx,
+			BallotThreshold: threshold,
+			BallotStatus:    types.BallotStatus_BallotInProgress,
+		}
+		k.SetBallot(ctx, &ballot)
+
+		isFinalized, isNew, ballot, _, err := k.VoteOnOutboundBallot(
+			ctx,
+			"index",
+			getValidEthChainIDWithIndex(t, 0),
+			common.ReceiveStatus_Success,
+			observer,
+		)
+		require.NoError(t, err)
+
+		// ballot should be finalized since there is only one observer
+		require.False(t, isFinalized)
+		require.False(t, isNew)
+		expectedBallot, found := k.GetBallot(ctx, "index")
+		require.True(t, found)
+		require.Equal(t, expectedBallot, ballot)
```

Comment:
maybe we can add a test in which the ballot is not finalized , but vote gets added . 
We can set the observer set to a higher number

Reply:
[580e897](https://github.com/zeta-chain/node/pull/1511/commits/580e897f52e5e630746686c06567784f9c00064a)